tests/exporters/test_s5.py

```diff
@@ -0,0 +1,308 @@
+        pressure_level = False
+
+        s5 = S5Exporter(
+            data_folder=tmp_path, granularity=granularity, pressure_level=pressure_level
+        )
+
+        expected_filepath = (
+            tmp_path / "raw/seasonal-monthly-single-levels/\
+            total_precipitation/2017_2018/M01_12.grib"
+        ).as_posix()
+        expected_filepath = expected_filepath.replace(' ', '')
+
+        variable = "total_precipitation"
+        max_leadtime = 5
+        min_year = 2017
+        max_year = 2018
+        min_month = 1
+        max_month = 12
+
+        processed_selection_request = s5.create_selection_request(
+            variable=variable,
+            max_leadtime=max_leadtime,
+            min_year=min_year,
+            max_year=max_year,
+            min_month=min_month,
+            max_month=max_month,
+        )
+
+        filepath = s5.make_filename(
+            dataset=s5.dataset, selection_request=processed_selection_request
+        )
+
+        assert (
+            expected_filepath == filepath.as_posix()
+        ), f"\
+            Expected: {expected_filepath}. Got: {filepath.as_posix()}"
+
+    @patch('cdsapi.Client')
+    def test_dataset_created_properly(self, tmp_path):
+
+        expected_datasets = [
+            "seasonal-monthly-single-levels",
+            "seasonal-monthly-pressure-levels",
+            "seasonal-original-single-levels",
+            "seasonal-original-pressure-levels",
+        ]
+        pressure_levels = [False, True, False, True]
+        granularities = ["monthly", "monthly", "hourly", "hourly"]
+
+        for ix, expected in enumerate(expected_datasets):
+            s5 = S5Exporter(
+                data_folder=tmp_path,
+                granularity=granularities[ix],
+                pressure_level=pressure_levels[ix],
+            )
+
+            assert (
+                s5.dataset == expected
+            ), f"\
+            Expected: {expected}. Got: {s5.dataset}"
+
+    @patch("cdsapi.Client")
+    def test_export_functionality(self, cdsapi_mock, tmp_path):
+        cdsapi_mock.return_value = Mock()
+        granularity = "monthly"
+        pressure_level = False
+
+        s5 = S5Exporter(
+            data_folder=tmp_path, granularity=granularity, pressure_level=pressure_level
+        )
+        s5.get_valid_variables()
+
+        variable = "total_precipitation"
+        min_year = 2017
+        max_year = 2017
+        min_month = 1
+        max_month = 1
+        max_leadtime = 1
+        N_parallel_requests = 1
+        show_api_request = True
+
+        s5.export(
+            variable=variable,
+            min_year=min_year,
+            max_year=max_year,
+            min_month=min_month,
+            max_month=max_month,
+            max_leadtime=max_leadtime,
+            show_api_request=show_api_request,
+            N_parallel_requests=N_parallel_requests,
+        )
+
+        (
+            tmp_path / "raw/seasonal-monthly-single-levels\
```

Comment:
What's happening here? Is an `assert` statement missing?

Reply:
@gabrieltseng i don't know how to test this because i don't actually want to download the file (because it will take like 24hrs + ... ) but i want to mock that behaviour

Reply:
Added an assert statement to make sure the cds api is called